db/local_db/local_client.go

```diff
@@ -248,3 +274,20 @@ func (c *LocalClient) getSchemaFromDB() (*sql.Rows, error) {
 	// we do NOT want to fetch the command schema
 	return c.dbClient.Query(fmt.Sprintf(query, constants.CommandSchema))
 }
+
+func establishConnection(connStr string) (*sql.DB, error) {
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		return nil, err
+	}
+	// limit to a single connection as we rely on session scoped data - temp tables and prepared statements
+	db.SetMaxOpenConns(1)
+	if db_common.WaitForConnection(db) {
+		return db, nil
+	}
+	return nil, fmt.Errorf("could not establish connection")
+}
+
+func (c *LocalClient) SetEnsureSessionStateFunc(f db_common.EnsureSessionStateCallback) {
```

Comment:
comment: 
```
// SetEnsureSessionStateFunc sets the callback function which setups the session data - prepared statements and introspection tables
// this will be called when there is a client error/timeout
```

Reply:
👍